internal/schema/schema.go

```diff
@@ -74,21 +77,49 @@ func NewEphemeral(resolver Resolver) Manager {
 	return mgr
 }
 
-func (m *manager) LoadSchema(ctx context.Context, url string) (*jsonschema.Schema, error) {
-	entry, ok := m.cache.Get(url)
+func (m *manager) LoadSchema(ctx context.Context, _url string) (*jsonschema.Schema, error) {
+	entry, ok := m.cache.Get(_url)
 	if ok {
 		return entry.schema, entry.err
 	}
 
 	e := &cacheEntry{}
-	e.schema, e.err = m.loadSchemaFromStore(ctx, url)
+	e.schema, e.err = m.loadSchemaFromStore(ctx, _url)
+	if e.err == nil || !errors.Is(e.err, fs.ErrNotExist) {
+		m.cache.Set(_url, e)
+		return e.schema, e.err
+	}
 
-	if e.err != nil && errors.Is(e.err, fs.ErrNotExist) {
-		e.err = fmt.Errorf("schema %q does not exist in the store", url)
+	var jsonSchemaErr *jsonschema.SchemaError
+	var pathErr *fs.PathError
+	if !errors.As(e.err, &jsonSchemaErr) || !errors.As(e.err, &pathErr) {
+		m.cache.Set(_url, e)
+		return e.schema, e.err
 	}
 
-	m.cache.Set(url, e)
-	return e.schema, e.err
+	parsedURL, err := url.Parse(_url)
+	if err != nil {
+		e.err = fmt.Errorf("failed to parse URL: %w", err)
+		m.cache.Set(_url, e)
+		return e.schema, e.err
+	}
+	if parsedURL.Scheme != URLScheme {
```

Comment:
We do support other schemes such as `http` and any unsupported schemes would be caught by https://github.com/cerbos/cerbos/blob/6b80dba8f39e7c8f8d7248902a4ae79eed460523/internal/schema/schema.go#L145